resources/hooks/useLogin.js

```diff
@@ -0,0 +1,46 @@
+import { useState } from "react";
+import { useNavigate } from "react-router";
+import { useUserStore } from "../js/useUserStore";
+import { apiUrl } from "../js/App";
+
+export const useLogin = () => {
+    const [errorMessage, setErrorMessage] = useState("");
+    let navigate = useNavigate();
+    const { setUser } = useUserStore();
+
+    const login = async (email, password) => {
+        // make request first to sanctum/csrf-cookie endpoint
+        //to initialize CSRF protection for the application
+        axios.get("/sanctum/csrf-cookie").then(() => {
+            const payload = {
+                email,
+                password,
+            };
+            axios
+                .post(`${apiUrl}/user/login`, payload, {
+                    headers: { Accept: "application/json" },
+                })
+                .then((response) => {
+                    console.log(response.data.user);
+                    if (response.data.user) {
+                        setErrorMessage(null);
+                        setUser(response.data.user);
+                        navigate("/");
+                    }
+                })
+                .catch((error) => {
+                    console.log(error);
+                    if (error.response) {
+                        if (!error.response.data.status) {
+                            setMessage("Email et/ou mot de passe incorrect(s)");
+                        }
+                    }
```

Comment:
Essaye ça voir si ça marche :)
```suggestion
                    if (!error.response?.data?.status) {
                        setMessage("Email et/ou mot de passe incorrect(s)");
                    }
```

Reply:
Wow! It is magic!